include/coro/detail/task_self_destroying.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <atomic>
+#include <coroutine>
+#include <cstdint>
+
+namespace coro::detail
+{
+
+class task_self_destroying;
+
+class promise_self_destroying
+{
+public:
+    promise_self_destroying();
+    ~promise_self_destroying();
+
+    promise_self_destroying(const promise_self_destroying&) = delete;
+    promise_self_destroying(promise_self_destroying&&);
+    auto operator=(const promise_self_destroying&) -> promise_self_destroying& = delete;
+    auto operator=(promise_self_destroying&&) -> promise_self_destroying&;
+
+    auto get_return_object() -> task_self_destroying;
+    auto initial_suspend() -> std::suspend_always;
+    auto final_suspend() noexcept -> std::suspend_never;
+    auto return_void() noexcept -> void;
+    auto unhandled_exception() -> void;
+
+    auto task_container_size(std::atomic<std::size_t>& task_container_size) -> void;
```

Comment:
Small suggestion: would `set_task_container_size_ref` sound more expressive?

Reply:
I could have sworn I read that the [https://isocpp.github.io/CppCoreGuidelines/](CppCoreGuidelines) suggested not having `get` and `set` prefixes on methods because its obvious that getters have a return type and setters take an argument and I've pretty much kept to that for this library. However I'm unable to find where this is, or if it ever was a thing... but you'll notice on most of the classes that is the style I've kept for this project.

e.g. `coro::task<>::handle() -> std::coroutine_handle<>` gets the underlying coroutine handle, but isn't named `get_handle()`.

I think I'd prefer to keep the consistency with how large this project is, but I appreciate the comment, hopefully the reasoning makes sense.

Reply:
Sounds good to me.